### PR TITLE
Fix deprecation warnings for chrono 0.4.23

### PR DESCRIPTION
**Pull request by dxtr85**

### CODE REVIEW

File: helix-term/tests/test/write.rs [hunk 1]
```diff
@@ -119,8 +119,9 @@ async fn test_write_fail_mod_flag() -> anyhow::Result<()> {
             (
                 Some(":w<ret>"),
                 Some(&|app| {
-                    assert_eq!(&Severity::Error, app.editor.get_status().unwrap().1);
-
+                    if let Some(status) = app.editor.get_status() {
+                        assert_eq!(&Severity::Error, status.1);
+                    }
```
Comment (the-mikedavis) [inline helix-term/tests/test/write.rs:124]: Looks like an unrelated change?

File: helix-core/src/increment/date_time.rs [hunk 2]
```diff
@@ -334,16 +334,19 @@ fn add_months(date_time: NaiveDateTime, amount: i64) -> Option<NaiveDateTime> {
...
     let ndays = ndays_in_month(year, date_time.month());
 
     if date_time.day() > ndays {
-        let d = NaiveDate::from_ymd(year, date_time.month(), ndays);
-        Some(d.succ().and_time(date_time.time()))
+        if let Some(d) = NaiveDate::from_ymd_opt(year, date_time.month(), ndays) {
+            d.succ_opt().map(|date| date.and_time(date_time.time()))
+        } else {
+            None
+        }
```
Comment (the-mikedavis) [inline helix-core/src/increment/date_time.rs:349]:
```suggestion
        NaiveDate::from_ymd_opt(year, date_time.month(), ndays)
            .and_then(|date| date.succ_opt())
            .map(|date| date.and_time(date_time.time()))
```

File: helix-term/tests/test/write.rs (hunk 1 above)
Comment (dxtr85) [reply helix-term/tests/test/write.rs:124]: I made this flaky test fix after [verification build](https://github.com/helix-editor/helix/actions/runs/3459194871) has failed on this test, so from my perspective it is related as without it this PR wouldn't have gone so far.
So I guess it depends on how you define related.

File: helix-core/src/increment/date_time.rs (hunk 2 above)
Comment (dxtr85) [reply helix-core/src/increment/date_time.rs]: Added in [this commit](https://github.com/helix-editor/helix/pull/4738/commits/3d623fdb093e048ddd42d143569625ccc3a90b5f).

**Comment by dxtr85** [comment]
@the-mikedavis I restored write.rs as suggested.

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks!
